base/channels.jl

```diff
@@ -218,6 +218,17 @@ end
 # does this right for you).
 isopen(c::Channel) = ((@atomic :acquire c.state) === :open)
 
+"""
+    empty!(c)
+
+Empty a Channel `c` by calling `empty!` on the internal buffer. 
+Returns the empty channel.
+"""
+function Base.empty!(c::Channel)
+    empty!(c.data)
```

Comment:
Hm, the concept of `empty!` seems a little strange for a Channel, since it is inherently somewhat race-y with new `put!` calls. But we can implement something of an optimized version of `while isready(c); take!(c); end` like so:
```suggestion
    @lock c begin
        ndrop = length(c.data)
        empty!(c.data)
        _increment_n_avail(c, -ndrop)
        notify(c.cond_put)
    end
```